Launcher.py

```diff
@@ -322,15 +322,28 @@ def main(args: Optional[Union[argparse.Namespace, dict]] = None):
     init_logging('Launcher')
     Utils.freeze_support()
     multiprocessing.set_start_method("spawn")  # if launched process uses kivy, fork won't work
-    parser = argparse.ArgumentParser(description='Archipelago Launcher')
+    parser = argparse.ArgumentParser(
+        description='Archipelago Launcher',
+        usage="[-h] [--update_settings] [Patch|Game|Component] [-- component args here]"
+        )
     run_group = parser.add_argument_group("Run")
     run_group.add_argument("--update_settings", action="store_true",
                            help="Update host.yaml and exit.")
     run_group.add_argument("Patch|Game|Component", type=str, nargs="?",
                            help="Pass either a patch file, a generated game or the name of a component to run.")
-    run_group.add_argument("args", nargs="*",
-                           help="Arguments to pass to component.")
-    main(parser.parse_args())
+    run_group.add_argument("--",
+                           help="Arguments to pass to component.",
+                           dest="args", default=[])
+    args = sys.argv[1:]
+    if "--" in args:
+        i = args.index("--")
+        passthrough_args = args[i+1:]
+        args = args[:i]
+        args = parser.parse_args(args)
+        args.args = passthrough_args
+    else:
+        args = parser.parse_args()
```

Comment:
Is this part necessary? `parser.parse_args` already seems to understand how to handle `--` correctly

Reply:
I think i figured out what the issue with this was (after reverting the `--` arg in the Parser), it joins args before and after the `--` arg together, so this happens
```
> python launcher.py test1, test2 -- test3 --test4
Namespace(update_settings=False, args=['test2', 'test3', '--test4'], **{'Patch|Game|Component': 'test1,'})
```

this is not a hill i'm willing to die on though, 

tested the rollback on these lines (and the 'args' arg) in source and frozen and the only scenarios that didn't work as expected were negative tests that just had invalid args ignored ex. `python launcher.py test test`

will commit if you confirm this is good enough:tm: and a worthy downside to get rid of the manual parsing

Reply:
got the confirmation in discord